prov/verbs/src/fi_verbs.h

```diff
@@ -112,6 +112,11 @@
 #define VERBS_DEF_CQ_SIZE 1024
 #define VERBS_MR_IOV_LIMIT 1
 
+#define VERBS_RX_CQ_DATA_IS_SET(info) 				\
+	(info && ((info->mode & FI_RX_CQ_DATA) || 		\
+		  (info->rx_attr && 				\
+		   (info->rx_attr->mode & FI_RX_CQ_DATA))))
```

Comment:
Let's make this generic and move to fi.h or fi_util.h.  Maybe something like (but more condensed if possible):
```
int ofi_check_rx_mode(info, flag)
{
    if (!info)
        return 0;
    if (info->rx_attr && info->rx_attr->mode)
        return info->rx_attr->mode & flag;
    return info->mode & flag;
}
```
This slightly changes the semantic from what you have.  If rx_attr->mode is non-zero it is used.  Otherwise info->mode is used.  This seems a reasonable interpretation of the man pages.

We can add checks to tx mode and rx/tx caps if needed.

Reply:
Sure.  👍